definitions/checks/foreman/check_checkpoint_segments.rb

```diff
@@ -0,0 +1,52 @@
+module Checks
+  module Foreman
+    class CheckpointSegments < ForemanMaintain::Check
+      metadata do
+        label :check_postgresql_checkpoint_segments
+        description 'Check if checkpoint_segments configuration exists on the system'
+        confine do
+          feature(:foreman) && feature(:installer) &&
+            File.exist?(feature(:installer).custom_hiera_file)
+        end
+      end
+
+      def run
+        failure_message = check_custom_hiera
+        fail! failure_message if failure_message
+      end
+
+      # rubocop:disable Metrics/MethodLength
+      def check_custom_hiera
+        hiera_file = feature(:installer) ? feature(:installer).custom_hiera_file : nil
```

Comment:
Only `hiera_file = feature(:installer).custom_hiera_file` is sufficient as you have already guarded check with feature and existence of file ?